### PR TITLE
Pipenv upgrade django to 3 (and prepare for 4)

### PR DESCRIPTION
**Pull request by raymondberg**
## Problem

Django 2.2 is already out of mainstream support and will soon be out of Extended Support.

## Solution

We should upgrade to the latest version...and get ready for Django 4 which will come out soon! 

### What is covered

Perform all updates up to 3.2. This has been tested locally and all of the key features seem to work.

I ran into some issues with later updates of some packages, so I did whatever was minimally required 
to stop the Deprecation warnings from running in `python -Wd` mode.

### What is not covered

Upgrading django_social is always a pain, so I skipped all pre-4.0 warnings there. https://github.com/chicagopython/chipy.org/issues/433

I also skipped norecaptcha-recaptcha https://github.com/chicagopython/chipy.org/issues/432

There are other deprecation warnings in tests that existed before I started but also relate to the two above. I'd like to get those two issues fixed before trying to clear up test deprecations.

This relies heavily on other PRs needing merging #429 and #431

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -22,9 +23,9 @@ WORKDIR ${SITE_DIR}
 RUN install -d proj/ var/log/ htdocs/ htdocs/static/
 
 # Install python packages
-ADD requirements.txt ${SITE_DIR}requirements.txt
-ADD requirements_test.txt ${SITE_DIR}requirements_test.txt
-RUN pip install -r ${SITE_DIR}requirements_test.txt
+ADD Pipfile ${SITE_DIR}Pipfile
+ADD Pipfile.lock ${SITE_DIR}
```
Comment (elmq0022) [inline docker/Dockerfile:27]: Quick question - Should the new line 27 be `Pipfile.lock ${SITE_DIR}Pipfile.lock`?  Looks odd compared to Pipfile and the old requirements.txt  and the requirements_test.txt.

**Review by elmq0022 — COMMENTED** [review]
Wow.  That was fast @raymondberg.  

So you didn't bump into any issues with incompatible packages?

File: docker/Dockerfile (hunk 1 above)
Comment (raymondberg) [reply docker/Dockerfile:27]: I'll fix this with a rebase

**Comment by raymondberg** [comment]
Nothing hard on this upgrade as far as I can tell. TinyMCE broke on later versions, but I got it up to 4.0 compatiblity

**Review by elmq0022 — APPROVED** [review]
Looks great! Thanks.

**Pull request merged**